core_lib/src/graphics/vector/vectorimage.cpp

```diff
@@ -82,6 +83,9 @@ bool VectorImage::read(QString filePath)
     {
         return false;
     }
+    ScopeGuard fileScope([&] {
+        file.close();
+    });
```

Comment:
you can use `OnScopeExit(file.close();)`

Reply:
BTW, QFile automatically closes the file upon destruction, so explicitly closing it is not necessary.

Reply:
Hmm interesting, where did you read that? I haven't been able to find anything that confirms that. I can only find the QFIleDevice::AutoCloseHandle which is DontCloseHandle by default, as such that should not apply here.

Anyway, I prefer that we do it explicitly :)

Reply:
yeh, that's not very obvious. it's in the description of the `~QFile()`. 
https://doc.qt.io/qt-6/qfile.html#dtor.QFile

If you look at the source code, it's in the destructor of `QFileDevice`, which is the base class of `QFile`
https://github.com/qt/qtbase/blob/4f9cbd5b2892f0f72c74db05d05c67f25e7c99e7/src/corelib/io/qfiledevice.cpp#L236

Reply:
If you can find a good place to call close, it's good to close file handle early as it can be released immediately.
however, if you put `close()` in a scope guard, it duplicates the behaviour already handled by QFile's destructor.